calico/etcddriver/hwm.py

```diff
@@ -192,6 +197,8 @@ def encode_key(key):
     operation should be a no-op on most keys but it's better to be tolerant
     here than to blow up.
     """
+    # PLW: is this going to be a painful performance hit? Agree with being
+    # defensive though, so probably got to live with it.
```

Comment:
Agree, it's annoying, but I think it's the right way to go.

I think it has a non-negligible perf hit but it's not the bottleneck.  (CPU went up when I added it but we're IO bound so it didn't affect throughput.)